skrf/network.py

```diff
@@ -429,6 +429,8 @@ def __init__(self, file: str = None, name: str = None, params: dict = None,
 
             # allows user to pass StringIO, filename or file obj
             if isinstance(file, io.StringIO):
+                if not hasattr(file, "name") and name is not None:
+                    file.name = name
```

Comment:
Is it essential to verify the format of the filename and ensure it conforms to the pattern `r"*.[ghsyz](\d+)p"`?

Reply:
The file name is verified inside the `Touchstone._parse_file` https://github.com/scikit-rf/scikit-rf/blob/master/skrf/io/touchstone.py#L375 function which is called afterwards, so here no verification necessary.